src/redis-cli.c

```diff
@@ -1356,11 +1363,18 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
         } else {
             /* Store database number when SELECT was successfully executed. */
             if (!strcasecmp(command,"select") && argc == 2 && config.last_cmd_type != REDIS_REPLY_ERROR) {
-                config.dbnum = atoi(argv[1]);
-                cliRefreshPrompt();
-            } else if (!strcasecmp(command,"auth") && (argc == 2 || argc == 3))
-            {
+                if (!config.in_multi){
+                    config.dbnum = atoi(argv[1]);
+                    cliRefreshPrompt();
+                }else{
+                    config.multi_dbnum = atoi(argv[1]);
+                }
```

Comment:
I think the cli should display the currently-selected db both in and outside multis, i.e.:

```c
config.dbnum = atoi(argv[1]);
if (config.in_multi) config.multi_dbnum = config.dbnum;
cliRefreshPrompt();
```